lib/OpenQA/BuildResults.pm

```diff
@@ -233,7 +236,8 @@ sub compute_build_results {
 
         while (my $job = $jobs->next) {
             $jr{distris}->{$job->DISTRI} = 1;
-            my $key = $job->TEST . "-" . $job->ARCH . "-" . $job->FLAVOR . "-" . $job->MACHINE;
+            my $test_name = $job_template_names->{$job->id} || $job->TEST;
```

Comment:
This would still be an additional db query per job, right?

Reply:
No, only one additional query for the whole computation. But like I said it won't scale because that query returns the job templates name globally for all jobs.